src/trace/trace.c

```diff
@@ -525,3 +525,22 @@ void trace_init(struct sof *sof)
 
 	dma_trace_init_early(sof);
 }
+
+void mtrace_dict_entry(uint32_t dict_entry_address, int n_args, ...)
+{
+	va_list ap;
+	int i;
+	char packet[MESSAGE_SIZE(4)];
+	uint32_t *args = (uint32_t *)&packet[MESSAGE_SIZE(0)];
+	const uint64_t tstamp = platform_safe_get_time(timer_get());
+
+	put_header((void *)packet, dt_tr.uuid_p, _TRACE_INV_ID, _TRACE_INV_ID,
```

Comment:
Because `packet` is a `char []`. I'll make the first argument of `put_header()` a `void *`, that requires no other change anywhere, not even in the `put_header()` code.